ctlog/ctlog.go

```diff
@@ -26,28 +26,56 @@ import (
 	ct "github.com/google/certificate-transparency-go"
 )
 
+// Interval represents the interval [Start, End).
+type Interval struct {
+	Start, End time.Time
+}
+
 // Log contains metadata about a CT Log that is needed by Monologue.
 type Log struct {
 	Name      string
 	URL       string
 	PublicKey crypto.PublicKey
 	MMD       time.Duration
+
+	// TemporalInterval represents the interval in which a certificate's
+	// NotAfter field must fall to be accepted by the Log (as specified by the
+	// Log Operators).
+	// TemporalInterval.Start and TemporalInterval.End are both to second
+	// precision.  Any smaller units may be ignored/discarded.
+	TemporalInterval *Interval
 }
 
 // New creates a Log structure, populating the fields appropriately.
 //
 // TODO(katjoyce): replace this implementation with something less hacky that
 // takes log details from a log list struct based on the new Log list JSON
 // schema.
-func New(url, name, b64PubKey string, mmd time.Duration) (*Log, error) {
+func New(url, name, b64PubKey string, mmd time.Duration, temporalStart, temporalEnd time.Time) (*Log, error) {
```

Comment:
Future consideration: If this gains more parameters, a builder pattern might be better, e.g.
```go
c := ctlog.New(url, name, pubkey, mmd).WithTemporalInterval(start, end)
```